test/e2e/sink_test.go

```diff
@@ -38,7 +38,7 @@ func TestSink(t *testing.T) {
 
 	// create broker
 	test.BrokerCreate(r, "default")
-	defer test.BrokerDelete(r, "default", true)
+	defer test.BrokerDelete(r, "default", false)
```

Comment:
There's a specific set of Broker tests that cover both --wait/--no-wait cases. Usually the flake appears in Sink or Trigger tests when used in `defer` as a cleanup. Even with added sleep to wait-loop I couldn't reproduce the behavior with  a consecutive create/delete cycles the events always got through. 

Since the defer is called after tests and finally the whole namespace is deleted, I tend to believe we don't need to wait for broker delete, nor cause the tests to fail falsely.